pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/BinaryProtoLookupService.java

```diff
@@ -0,0 +1,171 @@
+                                    return null;
+                                });
+                    } else {
+                        // (3) received correct broker to connect
+                        addressFuture.complete(responseBrokerAddress);
+                    }
+
+                } catch (Exception parseUrlException) {
+                    // Failed to parse url
+                    log.warn("[{}] invalid url {} : {}", destination.toString(), uri, parseUrlException.getMessage(),
+                            parseUrlException);
+                    addressFuture.completeExceptionally(parseUrlException);
+                }
+            }).exceptionally((sendException) -> {
+                // lookup failed
+                log.warn("[{}] failed to send lookup request : {}", destination.toString(), sendException.getMessage(),
+                        sendException);
+                addressFuture.completeExceptionally(sendException);
+                return null;
+            });
+        }).exceptionally(connectionException -> {
+            addressFuture.completeExceptionally(connectionException);
+            return null;
+        });
+        return addressFuture;
+    }
+    
+    
+    public CompletableFuture<PartitionedTopicMetadata> getPartitionedTopicMetadata(DestinationName destination) {
+
+        CompletableFuture<PartitionedTopicMetadata> partitionFuture = new CompletableFuture<PartitionedTopicMetadata>();
+
+        cnxPool.getConnection(serviceAddress).thenAccept(clientCnx -> {
+            long requestId = requestIdGenerator.getAndIncrement();
+            clientCnx.sendRequestWithId(Commands.newPartitionMetadataRequest(destination.toString(), requestId),
+                    requestId).thenAccept(partitionsStr -> {
+                        try {
+                            int partitions = Integer.parseInt(partitionsStr);
+                            partitionFuture.complete(new PartitionedTopicMetadata(partitions));
+                        } catch (Exception e) {
+                            partitionFuture.completeExceptionally(new PulsarClientException.LookupException(format(
+                                    "Failed to parse partition-response %s with %s", partitionsStr, e.getMessage())));
+                        }
+                    }).exceptionally((e) -> {
+                        log.warn("[{}] failed to get Partitioned metadata : {}", destination.toString(), e.getMessage(),
+                                e);
+                        partitionFuture.completeExceptionally(e);
+                        return null;
+                    });
+        }).exceptionally(connectionException -> {
+            partitionFuture.completeExceptionally(connectionException);
+            return null;
+        });
+
+        return partitionFuture;
+    }
+    
+    public String getServiceUrl() {
+    	return serviceAddress.toString();
+    }
+
+    static class LookupDataResult {
```

Comment:
```LookupResult``` POJO is being used to return Lookup result by Broker-http-lookup. However, to implement ```Lookup-Response-handling in binary-protocol lookup``` we need few more additional attributes such as ```authoritative, redirect, partitions```. So, we have this POJO at client side to handle ```Future.Response``` for ```topic-lookup``` and ```partition-lookup```